File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/achat/AchatMaterielAdminService.java
package ma.zs.stocky.service.facade.admin.achat;

import java.util.List;
import ma.zs.stocky.bean.core.achat.AchatMateriel;
import ma.zs.stocky.dao.criteria.core.achat.AchatMaterielCriteria;
import ma.zs.stocky.zynerator.service.IService;



public interface AchatMaterielAdminService extends  IService<AchatMateriel,AchatMaterielCriteria>  {

    List<AchatMateriel> findByBudgetId(Long id);
    int deleteByBudgetId(Long id);
    long countByBudgetId(Long id);
    List<AchatMateriel> findByTypeAchatMaterielId(Long id);
    int deleteByTypeAchatMaterielId(Long id);
    long countByTypeAchatMaterielId(Long id);




}

File: frontend/src/app/shared/service/admin/achat/AchatMaterielAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {AchatMaterielDto} from 'src/app/shared/model/achat/AchatMateriel.model';
import {AchatMaterielCriteria} from 'src/app/shared/criteria/achat/AchatMaterielCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class AchatMaterielAdminService extends AbstractService<AchatMaterielDto, AchatMaterielCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): AchatMaterielDto {
        return new AchatMaterielDto();
    }

    public constrcutCriteria(): AchatMaterielCriteria {
        return new AchatMaterielCriteria();
    }

    get API() {
        return environment.apiUrlStockyservice + 'admin/achatMateriel/';
    }
}
